lib/openhab/dsl/timers/timer.rb

```diff
@@ -92,6 +90,23 @@ def timer_block(semaphore)
             end
           }
         end
+
+        #
+        # Convert specific durations to something a timer understands, eg. items to seconds
+        #
+        # @params duration Duration
+        #
+        # @raise if duration cannot be used for a timer
+        #
+        # @return Converted duration if a conversion known, otherwise duration unchanged
+        #
+        def converted_duration(duration)
+          case duration
+          when nil, Java::JavaTimeTemporal::TemporalAmount then duration
+          when NumberItem then duration.to_f.seconds
```

Comment:
Yeah, this would be the most flexible approach and allow more items and even allow custom stuff (user supplied values, whatever they pass). I'll update it!

Reply:
I added a check for `#to_i`, too. Not sure though if there is a case where `#to_i` exists but `#to_f` not. WDYT?